DDF-1450 JUnit, Mockito, Powermock

ahoffer opened the pull request:
#### What does this PR do?
This PR updates several testing tools to current versions. 
- JUnit=4.12
- Mockito=1.10.19
- Powermock=1.6.4

#### Who is reviewing it?
@roelens8 @oconnormi @pklinef 

#### How should this be tested?
Full integration build tests. Look for new failures, errors, or warnings.

#### Any background context you want to provide?
In a related PR, the version of the testing tool hamcrest was updated.
XMLUnit and potentially other testing tools will be updates in a third and final PR after this PR is merged.

#### What are the relevant tickets?
DDF-1450

#### Screenshots (if appropriate)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

- XMLUnit and other testing tools will be included after this PR is merged
- Version of PowerMock earlier than 1.6.1 are not compatible with
  JUnit 4.12; upgrade PowerMock to 1.6.4
- Remove randomizing running from TestSolrFactory because of compile error
- Use mockito-core instead of mockito all to avoid including hamcrest and to resolve
- Exclude incompatible version of mockito from other libraries

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/619)
<!-- Reviewable:end -->

roelens8 commented on platform/migration/platform-migratable-util/pom.xml:
```diff
@@ -49,16 +49,22 @@
             <artifactId>powermock-api-mockito</artifactId>
             <version>${powermock.version}</version>
             <scope>test</scope>
+            <!--<exclusions>-->
```
Should these comments be removed?

roelens8 commented on libs/httpproxy/pom.xml:
```diff
@@ -77,13 +77,12 @@
-            <artifactId>mockito-all</artifactId>
-            <version>1.9.0</version>
+            <artifactId>mockito-core</artifactId>
+            <version>1.10.19</version>
```
Use dependency management here.

roelens8 commented on distribution/test/itests/pom.xml:
```diff
@@ -78,7 +78,7 @@
                 <groupId>junit</groupId>
                 <artifactId>junit</artifactId>
-                <version>4.11</version>
+                <version>4.12</version>
```
Use dependency management here.

roelens8 commented on distribution/test/pom.xml:
```diff
@@ -64,7 +64,7 @@
                 <groupId>junit</groupId>
                 <artifactId>junit</artifactId>
-                <version>4.11</version>
+                <version>4.12</version>
```
Use dependency management here.

tbatie replied:
Could use ${junit.version}

lessarderic replied:
Same comment as above.

roelens8 commented on libs/notifications/pom.xml:
```diff
@@ -93,7 +93,7 @@
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>1.9.5</version>
+            <version>1.10.19</version>
```
Use dependency management here.

ahoffer commented:
Review status: 0 of 26 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[distribution/test/itests/pom.xml, line 81 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSnfx-r1-81:-KCqbKhmUmuVSUM1gwtj:-499146022) ([raw file](https://github.com/codice/ddf/blob/493af66c4438ca4d80a740270abf117b262083c1/distribution/test/itests/pom.xml#L81)):*
Done!

---

*[libs/httpproxy/pom.xml, line 85 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSng--r1-85:-KCqbh0wEonKhkH537Ya:160415405) ([raw file](https://github.com/codice/ddf/blob/493af66c4438ca4d80a740270abf117b262083c1/libs/httpproxy/pom.xml#L85)):*
ok.

---

*[libs/notifications/pom.xml, line 96 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSng0-r1-96:-KCqbqL6GULbPnzCDO_v:5174689) ([raw file](https://github.com/codice/ddf/blob/493af66c4438ca4d80a740270abf117b262083c1/libs/notifications/pom.xml#L96)):*
ok

---

*[platform/migration/platform-migratable-util/pom.xml, line 55 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSng1-r1-55:-KCqbsAVF6yHU93e2Ime:442533232) ([raw file](https://github.com/codice/ddf/blob/493af66c4438ca4d80a740270abf117b262083c1/platform/migration/platform-migratable-util/pom.xml#L55)):*
I  thought I deleted those. I'll fix that.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/619)*
<!-- Sent from Reviewable.io -->

tbatie commented:
looks like we have a cfx.junit.version inside of the paltform-app pom.xml, doesn't look like we use it anywhere. Might be worth removing so no one uses the incorrect version.

tbatie commented on distribution/ddf-distro-repo/src/main/resources/commons-httpclient-3.1.pom:
```diff
@@ -1,254 +1,261 @@
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>3.8.1</version>
```
Any reason we aren't using the updated version here?

lessarderic commented on libs/notifications/pom.xml:
```diff
@@ -93,7 +95,7 @@
         <dependency>
```
Any reason we're adding `mockito-core` as a dependency here? I'd expect `mockito-all` to make this unnecessary.

lessarderic commented on libs/httpproxy/pom.xml:
```diff
@@ -77,13 +77,12 @@
             <groupId>org.mockito</groupId>
-            <artifactId>mockito-all</artifactId>
-            <version>1.9.0</version>
+            <artifactId>mockito-core</artifactId>
```
Any reason we're adding `mockito-core` as a dependency here? I'd expect `mockito-all` to make this unnecessary.

lessarderic commented on catalog/content/core/content-core-directorymonitor/pom.xml:
```diff
@@ -26,7 +26,6 @@
         <dependency>
```
Can't we just remove the whole `<dependency>` element?

lessarderic commented on distribution/test/itests/pom.xml:
```diff
@@ -78,7 +80,7 @@
                 <groupId>junit</groupId>
                 <artifactId>junit</artifactId>
-                <version>4.11</version>
+                <version>${junit.version}</version>
```
Since junit and hamcrest are already listed as dependencies in the root pom file, I don't think we need them here in the `<dependencyManagement>` section.

tbatie replied:
We were thinking the same thing, when we tried removing it ourselves it broke. Something isn't correct in our pom structure. I don't know if we need separate ticket to investigate this.

shaundmorris replied:
@tbatie do you recall what was broken? I completely removed hamcrest and junit and successfully ran all the itests but that was in master not on the branch here.

tbatie replied:
No idea, I only heard about it from others. I think @ahoffer tried it though, he may know. It'd be great if this was just a result from the changes in the pr though!

roelens8 commented:
Successfully hero'd PR. :+1:

ahoffer commented:
Removed. Is there some tool that will check for unreferenced properties? We might be able to do a bit more cleanup that way.

---

Review status: 0 of 26 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[catalog/content/core/content-core-directorymonitor/pom.xml, line 26 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzzsfocf8_89HOKV-r2-26:-KCvAWbWpvhgKNLKLFcm:712413511) ([raw file](https://github.com/codice/ddf/blob/690db6e516a0422df8b89887901c5893856fb6e5/catalog/content/core/content-core-directorymonitor/pom.xml#L26)):*
I removed it and I could recompile the entire core module, so I think it is safe to remove.

---

*[distribution/ddf-distro-repo/src/main/resources/commons-httpclient-3.1.pom, line 238 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSnfv-r2-238:-KCvArFKv54I4MPT2DLb:-1367601230) ([raw file](https://github.com/codice/ddf/blob/690db6e516a0422df8b89887901c5893856fb6e5/distribution/ddf-distro-repo/src/main/resources/commons-httpclient-3.1.pom#L238)):*
Updated to ${junit.version}

PS: That is the most unusual POM file I have seen

---

*[distribution/test/pom.xml, line 67 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSnfy-r2-67:-KCvBAAPeve7ilwV5Cst:1781875693) ([raw file](https://github.com/codice/ddf/blob/690db6e516a0422df8b89887901c5893856fb6e5/distribution/test/pom.xml#L67)):*
I'm not sure how I missed all of these.  It feels like a lost a commit somewhere along the way.

Updated junit and hamcrest dependencies to use maven properties.

---

*[distribution/test/itests/pom.xml, line 83 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSnfx-r2-83:-KCvCK6GZ4r_ebenSI4g:-2089210515) ([raw file](https://github.com/codice/ddf/blob/690db6e516a0422df8b89887901c5893856fb6e5/distribution/test/itests/pom.xml#L83)):*
Armand and I talked about adding the artifacts to a <dependencyManagement> section int the root POM. That way, sub-modules could omit the <version></version> element. I felt it would be a good change, but it would be out of scope of this ticket.

I have removed the dependencyManagement section from the test POM.

---

*[libs/httpproxy/pom.xml, line 85 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/619#-KCqJzztrlpVdXPpSng--r1-85:-KCvCrgR-dR0LorE8gdC:-1453678696) ([raw file](https://github.com/codice/ddf/blob/493af66c4438ca4d80a740270abf117b262083c1/libs/httpproxy/pom.xml#L85)):*
I misread your original comment. I changed it to mockito-all. Thanks.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/619#-:-KCv9r8poekPlCainof6:1239736250)*
<!-- Sent from Reviewable.io -->

shaundmorris commented:
@ahoffer did you see this on unreferenced properties? http://stackoverflow.com/questions/12420505/how-to-find-unused-properties-in-a-pom
